contrib/internal/httputil/trace_test.go

```diff
@@ -45,6 +45,38 @@ func TestTraceAndServe(t *testing.T) {
 		assert.Equal("service", span.Tag(ext.ServiceName))
 		assert.Equal("resource", span.Tag(ext.ResourceName))
 		assert.Equal("GET", span.Tag(ext.HTTPMethod))
+		assert.Equal(nil, span.Tag("http.host"))
+		assert.Equal("/", span.Tag(ext.HTTPURL))
+		assert.Equal("503", span.Tag(ext.HTTPCode))
+		assert.Equal("503: Service Unavailable", span.Tag(ext.Error).(error).Error())
+	})
+
+	t.Run("works with host", func(t *testing.T) {
```

Comment:
Instead of having this new text (seems like a lot of duplication), we can just sneak in a host into one of the other tests which create a request, like for example the `doubleStatus` one (last test).

Reply:
sure, we could do that, however having a hidden test in a check means that at some point this will not be tested if someone else refactors code and they don't know that we are specifically having host added to test some functionality. This will mean the test case can get lost during a refactor and no one will be any wiser.

since you maintain the repo, i'll let you decide which way you want to proceed.

Reply:
Thanks Roopak. I don't want us to take decisions because "_I maintain the repo_" unless that's really necessary. I think we should take the wisest decision together and I'm happy to discus.

I think you make a good point, but if we're going to do it like you say, let's change a few things and hopefully be patient with me while going through more iterations on this PR:

* Test names in Go are always one word by convention (I know other languages have sentences describing the test) which is more like an identifier of the test rather than a description. This will become more obvious when you run the tests with `-v` and notice that it replaces spaces with `_` further indicating that they are not meant to be sentences.
* Name the test `host`
* Have two sub-tests named `on` and `off` or `empty` and `ok` (or use a loop with two requests: empty and not empty)
* Remove assertions on the existence of this tag from other tests (since we're focusing them here as you suggest)
* Remove _everything_ from this test that is not strictly related to asserting the existence of `http.host`

What do you think?

Reply:
ok i took a shot at it. lmk what you think